src/routes/search.ts

```
@@ -0,0 +1,204 @@
+import { logWithContext, getCurrentCorrelationId } from '../services/logger.js';
+import { ErrorResponseSchema } from '../schemas/errors.js';
+import { SuccessResponseSchema } from '../schemas/common.js';
+
+/**
+ * Zod schema with enhanced security validation
+ */
+const searchRequestSchema = z.object({
+  query: z.string()
+    .min(1, 'Query cannot be empty')
+    .max(500, 'Query too long (maximum 500 characters)')
+    .regex(/^[^<>\x00-\x1f\x7f-\x9f]*$/, 'Query contains invalid characters'),
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)